test/functional/bulk.test.js

```diff
@@ -1625,6 +1628,58 @@ describe('Bulk', function () {
       });
   });
 
+  it('properly accounts for bson size in bytes in bulk ordered inserts', function () {
+    const client = this.configuration.newClient();
+    const size = MAX_BSON_SIZE / 2;
+    const largeString = crypto.randomBytes(size - 100).toString('hex');
+    const documents = [{ s: largeString }, { s: largeString }];
+
+    let db;
+
+    return client
+      .connect()
+      .then(() => {
+        db = client.db(this.configuration.db);
+        return db.dropCollection('doesnt_matter').catch(() => {});
+      })
+      .then(() => {
+        return db.createCollection('doesnt_matter');
+      })
+      .then(() => {
+        const coll = db.collection('doesnt_matter');
+        coll.insertMany(documents, { ordered: true });
+      })
+      .then(() => {
+        client.close();
+      });
```

Comment:
```suggestion
      .finally(() => client.close());
```
I think the errors on CI are related to the `client.close()` promise not being part of the chain (either simple arrow or a return is fine, no style preference). 

I suggest finally here incase there's an error the tests will still exit. We've just had a number of issues in the past with tests hanging the mocha run so we wanna make best effort going forward of making sure we get a clean exit whenever we can.

Reply:
Updated.